File: e2e/elixir/test/rate_limit_test.exs
# E2e tests for category: rate_limit
defmodule E2e.RateLimitTest do
  use ExUnit.Case, async: true

  describe "rate_limit_basic_delay" do
    test "Rate limiter adds delay between requests to the same domain" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 3
      assert result.rate_limit.min_duration_ms >= 150
    end
  end

  describe "rate_limit_zero_no_delay" do
    test "Rate limiter with zero delay does not slow crawling" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 2
    end
  end
end

File: e2e/rust/tests/rate_limit_test.rs
//! E2e tests for category: rate_limit

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[tokio::test]
async fn test_rate_limit_basic_delay() {
    // Rate limiter adds delay between requests to the same domain
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'crawl.pages_crawled' not available on result type
    // skipped: field 'rate_limit.min_duration_ms' not available on result type
}

#[tokio::test]
async fn test_rate_limit_zero_no_delay() {
    // Rate limiter with zero delay does not slow crawling
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'crawl.pages_crawled' not available on result type
}
